src/lib/helpers.ts

```diff
@@ -121,17 +121,13 @@ export class Deferred<T = any> {
   }
 }
 // Taken from: https://stackoverflow.com/questions/38552003/how-to-decode-jwt-token-in-javascript-without-using-a-library
-export function decodeJWT(token: string) {
-  const base64Url = token.split('.')[1]
-  const base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/')
-  const jsonPayload = decodeURIComponent(
-    atob(base64)
-      .split('')
-      .map(function (c) {
-        return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2)
-      })
-      .join('')
-  )
+export function decodeJWTPayload(token: string) {
```

Comment:
Any idea when this will be merged? I'm working on a pull request for something else, and was gonna use a similar decode function. The code I pulled this morning didn't have one, so I was gonna add it. But this PR negates the need.

Reply:
Also, I was gonna use this function to verify that a passed-in `access_token` was quasi-valid. In other words, folks couldn't just pass an empty string or `hello.there.world` into the method I planned on adding.

So if you wanted to be slightly more paranoid, you could do this:
```js
const base64regex = /^([0-9a-zA-Z+/]{4})*(([0-9a-zA-Z+/]{2}==)|([0-9a-zA-Z+/]{3}=))?$/
const parts = token.split('.')

if (parts.length !== 3) {
  throw new Error('JWT is not valid: not a JWT structure')
}

if (!base64regex.test(parts[1])) {
  throw new Error('JWT is not valid: not a JWT structure')
}
```

Reply:
Thanks! Feel free to pull this out in another PR, and we'll merge it!

Reply:
Also the regular expression is somewhat wrong as JWTs use Base64-URL. This is a correct regular expression:

```javascript
/^([a-z0-9_-]{4})*($|[a-z0-9_-]{3}=?$|[a-z0-9_-]{2}(==)?$)$/i
```